superset-frontend/src/dashboard/util/charts/getEffectiveExtraFilters.ts

```diff
@@ -19,9 +19,11 @@
 import { DataRecordFilters } from '@superset-ui/chart';
 
 export default function getEffectiveExtraFilters(filters: DataRecordFilters) {
-  return Object.entries(filters).map(([column, values]) => ({
-    col: column,
-    op: 'in',
-    val: values,
-  }));
+  return Object.entries(filters)
+    .map(([column, values]) => ({
+      col: column,
+      op: Array.isArray(values) ? 'IN' : '=',
+      val: values,
+    }))
+    .filter(filter => filter.val !== null);
```

Comment:
when user clear out a filter, it's value will be `null`, and we need to send out this to backend.

Reply:
@graceguo-supercat when the filter is cleared, the line will be removed, thus triggering a filter event (tested that it works properly on both legacy and new charts). Having an undefined value for a binary or set filter is also invalid according to the type declarations on `superset-ui`, so if this is the case we probably need to make changes there.

Reply:
agree!